avaframe/in3Utils/getReleaseArea.py

```diff
@@ -0,0 +1,247 @@
+    yv = np.arange(-0.5 * y_end, 0.5 * y_end, dx)
+    hr = float(cfgGen['hr'])
+    vol = float(cfgGen['vol'])
+    dh = float(cfgGen['dh'])
+    lenp = int(cfgGen['lenP'])
+    alpha_stop = float(cfgHS['alpha_stop'])
+
+    # get A, B from HS
+    [A, B, f_len] = gT.getParams(cfgTopo)
+
+    # Compute release area ---------------------------------------------------------------------------------------
+    # along valley margin of release area at alpha_stop° point
+    xStop = (np.tan(np.radians(-alpha_stop)) - B) / (2. * A)
+    xr = hr / np.sin(np.radians(alpha_stop))
+    xrp = hr / np.tan(np.radians(alpha_stop))
+    yr = vol / (xr * dh)
+
+    # define corner coordinates
+    xyPoints = np.zeros((lenp, 2))
+    xyPoints[0, 0] = xStop - xrp        # xll
+    xyPoints[0, 1] = -0.5 * yr           # yll
+    xyPoints[1, 0] = xStop              # xlr
+    xyPoints[1, 1] = -0.5 * yr          # ylr
+    xyPoints[2, 0] = xStop             # xur
+    xyPoints[2, 1] = 0.5 * yr           # yur
+    xyPoints[3, 0] = xStop - xrp       # xul
+    xyPoints[3, 1] = 0.5 * yr           # yul
+
+    # list all points
+    for m in range(lenp):
+        log.info('Point %d: x %f y %f' % (m, xyPoints[m, 0], xyPoints[m, 1]))
+
+    return xv, yv, xyPoints
+
+
+def correctOrigin(xv, yv, xyPoints, cfgT, y_end):
+    """ Move the points so that the correct origin is set """
+
+    xl = float(cfgT['DEMDATA']['xl'])
+    yl = float(cfgT['DEMDATA']['yl'])
+
+    xv = xv + xl
+    yv = yv + yl + 0.5 * y_end
+
+    xyPoints[:,0] = xyPoints[:,0] + xl
+    xyPoints[:,1] = xyPoints[:,1] + yl + 0.5 * y_end
+
+    return xv, yv, xyPoints
+
+
+def writeNXYZ(xyPoints, DEM_type, cfgFile, cfgGen, outDir):
+    """ Write topography information to file """
+
+    lenp = len(xyPoints)
+    rel_no = int(cfgFile['rel_no'])
+    rel_h = float(cfgGen['dh'])
+    rel_name = cfgFile['rel_name']
+
+    # Add vertical coordinate
+    z = np.zeros((lenp, 1))
+    p_mat = np.matrix(np.append(xyPoints, z, axis=1))
+
+    # Save elevation data to .asc file and add header lines
+    with open(os.path.join(outDir, 'release_%d%s.nxyz' % (rel_no, DEM_type)), 'w') as f:
+        f.write('name=%s\n' % (rel_name))
+        f.write('ah=%.2f\n' % (rel_h))
+        f.write('rho=None\n')
+        f.write('%d\n' % (lenp))
+        for line in p_mat:
+            np.savetxt(f, line, fmt='%f')
+
+    # Log info here
+    log.info('Release Area written to: %s/release_%d%s.nxyz' % (outDir, rel_no, DEM_type))
+    if cfgGen.getboolean('outputtxt'):
+        shutil.copyfile(os.path.join(outDir, 'release_%d%s.nxyz' % (rel_no, DEM_type)),
+        os.path.join(outDir, 'release_%d%s.txt' % (rel_no, DEM_type)))
+
+
+def makeOutputDir(avalancheDir):
+    """ Make output directory """
+
+    outDir = os.path.join(avalancheDir, 'Inputs', 'REL')
+    if os.path.isdir(outDir):
+        log.warning('Be careful %s already existed - new data is added' % (outDir))
+    else:
+        os.makedirs(outDir)
+
+    return outDir
```

Comment:
See question on makeOutputDir for topography..
Also: could this be solved with one makeOutputDir function?

Reply:
Now outDir is set directly